packages/bluesnap-direct-integration/src/bluesnap-direct-script-loader.spec.ts

```diff
@@ -0,0 +1,56 @@
+import { createScriptLoader, ScriptLoader } from '@bigcommerce/script-loader';
+
+import { PaymentMethodClientUnavailableError } from '@bigcommerce/checkout-sdk/payment-integration-api';
+
+import BlueSnapDirectScriptLoader, { BlueSnapDirectSdkEnv } from './bluesnap-direct-script-loader';
+import { BlueSnapDirectSdk } from './types';
+
+describe('BlueSnapDirectScriptLoader', () => {
+    let scriptLoader: ScriptLoader;
+    let blueSnapDirectScriptLoader: BlueSnapDirectScriptLoader;
+    let blueSnapDirectSdkMock: BlueSnapDirectSdk;
+
+    beforeEach(() => {
+        scriptLoader = createScriptLoader();
+        blueSnapDirectScriptLoader = new BlueSnapDirectScriptLoader(scriptLoader);
+        blueSnapDirectSdkMock = {} as BlueSnapDirectSdk;
+
+        jest.spyOn(scriptLoader, 'loadScript').mockImplementation(() => {
+            Reflect.set(window, 'bluesnap', blueSnapDirectSdkMock);
+
+            return Promise.resolve();
+        });
+    });
+
+    describe('#load', () => {
+        it('should load the Hosted Payment Fields SDK successfully', async () => {
+            const sdk = await blueSnapDirectScriptLoader.load();
+
+            expect(sdk).toEqual(blueSnapDirectSdkMock);
+        });
+
+        it('should load the Hosted Payment Fields SDK for production', async () => {
+            await blueSnapDirectScriptLoader.load();
+
+            expect(scriptLoader.loadScript).toHaveBeenCalledWith(BlueSnapDirectSdkEnv.PRODUCTION);
+        });
+
+        it('should load the Hosted Payment Fields SDK for sandbox', async () => {
+            await blueSnapDirectScriptLoader.load(true);
+
+            expect(scriptLoader.loadScript).toHaveBeenCalledWith(BlueSnapDirectSdkEnv.SANDBOX);
+        });
+
+        it('should fail to load the Hosted Payment Fields SDK', async () => {
+            jest.spyOn(scriptLoader, 'loadScript').mockImplementationOnce(() => {
+                Reflect.deleteProperty(window, 'bluesnap');
```

Comment:
Do we really need to use Reflect here, and not use object methods?

Reply:
I need to delete de prop to simulate that the script was not loaded. `delete` operator is not allowed, the linter complains.

Reply:
You may export a window interface from `is-bluesnap-direct-window.ts` as:
```
export interface BlueSnapDirectHostWindow extends Window {
    bluesnap?: BlueSnapDirectSdk;
}
```
In this file, at beginning:
```
jest.spyOn(scriptLoader, 'loadScript').mockImplementation(() => {
    (window as BlueSnapDirectHostWindow).bluesnap = blueSnapDirectSdkMock;

    return Promise.resolve();
});
```
And here:
```
jest.spyOn(scriptLoader, 'loadScript').mockImplementationOnce(() => {
    delete (window as BlueSnapDirectHostWindow).bluesnap;

    return Promise.resolve();
});
```

Reply:
Done.

Reply:
Undone.